main.go

```diff
@@ -0,0 +1,127 @@
+package authenticated_block_store
+
+import (
+	"fmt"
+	"log"
+	"io/ioutil"
+        "strings"
+        "os"
+        "crypto/sha256"
+        "math/rand"
+        "time"
+
+)
+
+//size of each block, intended to be 4KB TODO
+const BLOCK_SIZE = 16
+
+var (
+        logFileNames []string
+        logFileHandles []*os.File
+)
+
+type logFileNode struct {
+        name string
+        hash string
+        nextHash string
+        next *logFileNode
+        previous *logFileNode //might not be needed
+}
+
+type dataBase struct {
+        head *logFileNode
+        tail *logFileNode
+        merkleRoot string
+        // other components
+}
+
+func initializeDB() *dataBase {
+        return &dataBase{
+                head:nil,
+                tail:nil,
+                merkleRoot:"",
+        }
+}
+
+func readLogFile(input *os.File) *logFileNode{
+        //do sanity checks
+        var newNode logFileNode
+        var err error
+        newNode.name = input.Name()
+        b := make ([]byte, 4)
+        _, err = input.ReadAt(b, 0)
+        check (err)
+        newNode.hash = string(b)
+        _, err = input.ReadAt(b, 0)
+        check (err)
+        return &newNode
+}
+
+func check(e error) {
+    if e != nil {
+        panic(e)
+    }
+}
+
+func randomGen() *rand.Rand {
+        seed := rand.NewSource(time.Now().UnixNano())
+        random := rand.New(seed)
+        return random
+}
+// generate a random block of data
+func getBlock() []byte {
+        newData := make ([]byte, BLOCK_SIZE)
+        myRand := randomGen()
+        for index, _ := range(newData) {
+                newData[index] = byte(myRand.Intn(256))
+        }
+        fmt.Println("New random dataBlock generated:", newData)
+        return newData
+}
+
+
+func main() {
+        fmt.Println("Test main function.")
+	myDB, err:= Create()
+	if (err != nil) {
+            log.Fatal(err)
+        }
+	fmt.Println("DUMMY:", myDB)
+        logFilesPath := "."
+        logFilesSuffix := ".log" 
+
+        allFiles, err := ioutil.ReadDir(logFilesPath)
+        check(err)
+
+        for _, tmpFile := range allFiles {
+//              fmt.Println(tmpFile.Name())
```

Comment:
Indenting is weird here too. See if gofmt fixes.

Reply:
Yup, going to use that on all files, thanks.
(Initially it looked better on my machine!)